src/library/autodj/autodjprocessor.cpp

```diff
@@ -859,17 +855,28 @@ void AutoDJProcessor::playerPositionChanged(DeckAttributes* pAttributes,
             double transitionProgress = (thisPlayPosition - thisDeck->fadeBeginPos) /
                     (thisDeck->fadeEndPos - thisDeck->fadeBeginPos);
             double transitionStep = transitionProgress - m_transitionProgress;
+
+            if ((transitionProgress == 0.0) &&
+                    (m_crossfaderStartCenter)) {
+                setCrossfader(0.0); // So we start at the mid point before we make any fader moves.
+            }
             if (transitionStep > 0.0) {
                 // We have made progress.
                 // Backward seeks pause the transitions; forward seeks speed up
                 // the transitions. If there has been a seek beyond endPos, end
                 // the transition immediately.
-                double remainingCrossfader = crossfaderTarget - currentCrossfader;
-                double adjustment = remainingCrossfader /
-                        (1.0 - m_transitionProgress) * transitionStep;
-                // we move the crossfader linearly with
-                // movements in this track's play position.
-                setCrossfader(currentCrossfader + adjustment);
+                if ((m_crossfaderStartCenter) &&
+                        (transitionProgress <= ((thisDeck->fadeEndPos - thisDeck->fadeBeginPos) / 2))) {
+                    // We want the cross fader to remain in the middle.
```

Comment:
It'd be good to explain the purpose here.
Something along those lines, feel free to edit.
```suggestion
                    // For the first half of the transition we want the "from" deck
                    // to play at full volume like with a full crossfade transition,
                    // hence we keep the crossfader in the middle.
```

Reply:
Sure.   Note that ...what happens is the fader moves to the mid, and then the transition from mid to the end position takes the full duration of the change (which is much neater than my intent!).  This code (and comment)  was to make sure that a re-calculation of 'transitionProgress' didn't drop us back to (near) the start.